src/main/java/org/broadinstitute/hellbender/tools/dataflow/pipelines/BaseRecalibratorDataflow2.java

```diff
@@ -0,0 +1,282 @@
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.baq.BAQ;
+import org.broadinstitute.hellbender.utils.dataflow.BucketUtils;
+import org.broadinstitute.hellbender.utils.dataflow.DataflowUtils;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * First pass of the base quality score recalibration -- Generates recalibration table based on various covariates
+ * (such as read group, reported quality score, machine cycle, and nucleotide context).
+ * <p>
+ * This version uses the new skeleton transform.
+ * <p>
+ * Dataflow version of BaseRecalibrator.
+ */
+@CommandLineProgramProperties(
+        summary = "First pass of the Base Quality Score Recalibration (BQSR) -- Generates recalibration table based on various user-specified covariates (such as read group, reported quality score, machine cycle, and nucleotide context).",
+        oneLineSummary = "Generates recalibration table",
+        programGroup = ReadProgramGroup.class
+)
+public class BaseRecalibratorDataflow2 extends DataflowCommandLineProgram implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    /**
+      * Reference window function for BQSR. For each read, returns an interval representing the span of
+      * reference bases required by the BQSR algorithm for that read. Should be passed into the
+      * {@link RefAPIMetadata} object for the {@link AddContextDataToRead} transform.
+      */
+    public static final SerializableFunction<GATKRead, SimpleInterval> BQSR_REFERENCE_WINDOW_FUNCTION =
+            read -> BAQ.getReferenceWindowForRead(read, BAQ.DEFAULT_BANDWIDTH, BAQ.DEFAULT_INCLUDE_CLIPPED_BASES);
+
+    public static final String NO_DBSNP_EXCEPTION = "This calculation is critically dependent on being able to skip over known variant sites. Please provide a VCF file containing known sites of genetic variation.";
+
+    private final static Logger logger = LogManager.getLogger(BaseRecalibratorDataflow2.class);
+    // temporary file with the serialized recalibrationTables.
+    private final static String TEMP_RECALTABLES = "temp-ds-recaltables";
+
+    // ------------------------------------------
+    // Command-line options
+
+    /**
+     * all the command line arguments for BQSR and its covariates
+     */
+    @ArgumentCollection(doc = "all the command line arguments for BQSR and its covariates")
+    private transient final BaseRecalibrationArgumentCollection BRAC = new BaseRecalibrationArgumentCollection();
+
+    @Argument(doc = "the reference name (on GG)",
+        fullName = "apiref", optional = false)
+    protected String referenceURL;
```

Comment:
Perhaps you should be using the existing `BRAC.referenceArguments` argument to pass in this value (otherwise this tool will have two reference arguments, one of which will be silently ignored). 

Alternatively, you could check `BRAC.referenceArguments` explicitly in `setupPipeline()` and, if it has a value, throw an error instructing the user to use `--apiref` instead.